### PR TITLE
feat(planner): plan window function

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

initial impl of window function plan.

Closes #issue

### CODE REVIEW

File: src/query/sql/src/planner/binder/aggregate.rs [hunk 1]
```diff
@@ -55,6 +57,9 @@ pub struct AggregateInfo {
     /// Arguments of aggregation functions
     pub aggregate_arguments: Vec<ScalarItem>,
 
+    /// infos for window with aggregate function
+    pub window_info: Option<WindowInfo>,
```
Comment (leiysky) [inline src/query/sql/src/planner/binder/aggregate.rs:61]: There will be multiple windows if there are window functions with different partition definitions? For example:
```sql
SELECT count(*) over(partition by a), count(*) over(partition by b) FROM t
```

File: src/query/sql/src/planner/binder/window.rs [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+                                ),
+                            });
+                            found = true;
+                            break;
+                        }
+                    }
+
+                    if found {
+                        continue;
+                    }
```
Comment (leiysky) [inline src/query/sql/src/planner/binder/window.rs:129]: I think we don't need to do this for `order by` in window functions. It's different from `order by` clause in the query body, we can just throw an error if no matched column is found.

File: src/query/sql/src/planner/binder/aggregate.rs [hunk 3]
```diff
@@ -152,6 +152,8 @@ impl<'a> AggregateRewriter<'a> {
             ScalarExpr::SubqueryExpr(_) => Ok(scalar.clone()),
 
             ScalarExpr::AggregateFunction(agg_func) => self.replace_aggregate_function(agg_func),
+
+            ScalarExpr::WindowFunction(_) => unreachable!(),
```
Comment (xudong963) [inline src/query/sql/src/planner/binder/aggregate.rs:156]: better to return err info

**Comment by ariesdevil** [comment]
If there are no obvious problems, let's merge this pr first and I'll move on to the next.

**Review by xudong963 — APPROVED** [review]

**Review by sundy-li — APPROVED** [review]

**Pull request merged**